test/python/unit/api/test_sort.py

```diff
@@ -0,0 +1,714 @@
+#####################
+## test_sort.py
+## Top contributors (to current version):
+##   Yoni Zohar
+## This file is part of the CVC4 project.
+## Copyright (c) 2009-2021 by the authors listed in the file AUTHORS
+## in the top-level source directory and their institutional affiliations.
+## All rights reserved.  See the file COPYING in the top-level source
+## directory for licensing information.
+##
+## ########################################################
+##
+## Unit tests for sort API.
+## Obtained by translating test/unit/api/sort_black.cpp
+
+import pytest
+import pycvc4
+from pycvc4 import kinds
+from pycvc4 import Sort
+
+
+@pytest.fixture
+def solver():
+    return pycvc4.Solver()
+
+
+def create_datatype_sort(solver):
+    intSort = solver.getIntegerSort()
+    # create datatype sort to test
+    dtypeSpec = solver.mkDatatypeDecl("list")
+    cons = solver.mkDatatypeConstructorDecl("cons")
+    cons.addSelector("head", intSort)
+    cons.addSelectorSelf("tail")
+    dtypeSpec.addConstructor(cons)
+    nil = solver.mkDatatypeConstructorDecl("nil")
+    dtypeSpec.addConstructor(nil)
+    dtypeSort = solver.mkDatatypeSort(dtypeSpec)
+    return dtypeSort
+
+
+def create_param_datatype_sort(solver):
+    sort = solver.mkParamSort("T")
+    paramDtypeSpec = solver.mkDatatypeDecl("paramlist", sort)
+    paramCons = solver.mkDatatypeConstructorDecl("cons")
+    paramNil = solver.mkDatatypeConstructorDecl("nil")
+    paramCons.addSelector("head", sort)
+    paramDtypeSpec.addConstructor(paramCons)
+    paramDtypeSpec.addConstructor(paramNil)
+    paramDtypeSort = solver.mkDatatypeSort(paramDtypeSpec)
+    return paramDtypeSort
+
+
+def test_operators_comparison(solver):
+    try:
+        solver.getIntegerSort() == Sort(solver)
+        solver.getIntegerSort() != Sort(solver)
+        solver.getIntegerSort() < Sort(solver)
+        solver.getIntegerSort() <= Sort(solver)
+        solver.getIntegerSort() > Sort(solver)
+        solver.getIntegerSort() >= Sort(solver)
+    except RuntimeError:
```

Comment:
Quick question: is there a reason to use the `try` `except` as opposed to just asserting each of those conditions individually? It should fail if any of them throw an exception, right? Is it because you want to abort the whole test?

Reply:
I have two weak reasons:
1. I want to distinguish between `RuntimeError` (which is thrown from the API if something bad happens) and other errors like syntax errors, typos, etc. 
2. I want to be similar to the cpp api tests

If you think this is unnecessary, I have no problem removing all these `try`s. 

BTW, @Ying1123  is preparing a PR for the solver_black tests in which she has implemented a `not_throws` function:
https://github.com/Ying1123/CVC4/blob/3770e0edcd93e4659867d5dcba7557d3f5e8d3b4/test/python/unit/api/test_solver.py#L19
used like this:
https://github.com/Ying1123/CVC4/blob/3770e0edcd93e4659867d5dcba7557d3f5e8d3b4/test/python/unit/api/test_solver.py#L44

having this would make us even more similar to the cpp api tests. If we decide to go with these checks, I can either bring her function to this PR, or refactor the tests after her code gets merged. What do u think?

Reply:
Cool, thanks for the explanation! I don't feel super strongly either way. I don't think matching it super closely to C++ is necessary, although there is value in that. There's a bit of tension there between having very similar APIs and allowing Pythonic use. As for `assert_not_throws`, I think it could make sense to have a function like that. I don't quite follow what the `ExpectedException` argument is for though. Because if it's not supposed to throw, why would there be an expected exception?

Reply:
this is again the distinction between `RuntimeError` and other exceptions. `RuntimeError` will go to the `pytest.fail()` line while other exceptions will fail in the failed line, which will create a difference.

It would be simpler, of course, to just write the commands, and if they fail then they fail. The code will be more readable, maybe even `pythonic` (whatever that means :) ) but just different then cpp, that i see you don't think is very important. @Ying1123 was also advocating removing these `try` statements. So why don't we just drop them?

Reply:
dropped here: 29a0821
Would be interested in knowing what u think. Of course, this commit can be reverted.

Reply:
Yeah I think that looks good! I'd vote to stick with that but as mentioned earlier don't mind if you'd rather have some extra error infrastructure like the `assert_not_throws`.